lib/archethic/account/mem_tables/token_ledger.ex

```diff
@@ -142,22 +173,33 @@ defmodule Archethic.Account.MemTables.TokenLedger do
   """
   @spec get_inputs(binary()) :: list(VersionedTransactionInput.t())
   def get_inputs(address) when is_binary(address) do
-    @unspent_output_index_table
-    |> :ets.lookup(address)
-    |> Enum.map(fn {_, from, token_address, token_id} ->
-      [{_, amount, spent?, timestamp, protocol_version}] =
-        :ets.lookup(@ledger_table, {address, from, token_address, token_id})
-
-      %VersionedTransactionInput{
-        input: %TransactionInput{
-          from: from,
-          amount: amount,
-          type: {:token, token_address, token_id},
-          spent?: spent?,
-          timestamp: timestamp
-        },
-        protocol_version: protocol_version
-      }
-    end)
+    unspent =
+      @unspent_output_index_table
+      |> :ets.lookup(address)
+      |> Enum.reduce([], fn {_, from, token_address, token_id}, acc ->
+        case :ets.lookup(@ledger_table, {address, from, token_address, token_id}) do
+          [] ->
+            acc
+
+          [{_, amount, spent?, timestamp, protocol_version}] ->
+            [
+              %VersionedTransactionInput{
+                input: %TransactionInput{
+                  from: from,
+                  amount: amount,
+                  type: {:token, token_address, token_id},
+                  spent?: spent?,
+                  timestamp: timestamp
+                },
+                protocol_version: protocol_version
+              }
+              | acc
+            ]
+        end
+      end)
+      |> Enum.reverse()
+
+    spent = Inputs.get_inputs(:token, address)
+    Enum.concat(spent, unspent)
```

Comment:
You could pass the spent inputs as the reduce's accumulator. So, you could prepend to the list the unspent output in the reduce's function which will be more efficient than concatenating the two lists.